src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.SetTimes.OSX.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    internal partial struct FileStatus
+    {
+        internal void SetCreationTime(string path, DateTimeOffset time)
+        {
+            // Try to set the attribute on the file system entry using setattrlist,
+            // if we get ENOTSUP then it means that "The volume does not support
+            // setattrlist()", so we fall back to the method used on other unix
+            // platforms, otherwise we throw an error if we get one, or invalidate
+            // the cache if successful because otherwise it has invalid information.
+            Interop.Error error = SetCreationTimeCore(path, time);
+
+            if (error == Interop.Error.ENOTSUP)
+            {
+                SetAccessOrWriteTimeCore(path, time, isAccessTime: false, checkCreationTime: false);
+            }
```

Comment:
Do you have a test where you verify that this code path is actually hitting?

Reply:
No I do not, I have no idea what the specific conditions for this failing is (ie. which 'volume' type), and it would require a weird volume type anyway probably. I'll add a comment to the comment above that states that it should be changed with care since there are no tests for it.

Reply:
I think we should investigate which specific scenario returns ENOTSUP and add a unit test for it, unless is super hard to setup the scenario.